File: examples/equos-nodejs-integration/src/run-session.ts
import "dotenv/config";

import {
  Equos,
  type CreateEquosSessionResponse,
  type EquosSession,
} from "@equos/node-sdk";

export const runSession = async () => {
  const client = Equos.client(process.env.EQUOS_API_KEY!);

  const session: CreateEquosSessionResponse = await client.sessions.create({
    name: "Translation session",
    client: "user...", // Optional: your end-user identifier, use full for resource segmentation.
    agent: { id: process.env.EQUOS_AGENT_ID! },
    avatar: { id: process.env.EQUOS_AVATAR_ID! },
    consumerIdentity: {
      name: "Your user name",
      identity: "your-user-id",
    },
  });

  console.log("Session ID:", session.session.id);
  console.log("Session Status:", session.session.status);
  console.log("Session Consumer AccessToken:", session.consumerAccessToken);

  await new Promise((resolve) => setTimeout(resolve, 10000));

  // Do not forget to stop the session when done.
  const ended: EquosSession = await client.sessions.stop(session.session.id);
  console.log("Ended Session Status:", ended.status);
};

runSession();

File: examples/equos-python-integration/src/python_integration/run_session.py
import os

from typing import Optional

from equos import (
    Equos,
    EquosResourceId,
    CreateEquosSessionRequest,
    CreateEquosSessionResponse,
    EquosParticipantIdentity,
    EquosSession,
)

from dotenv import load_dotenv

load_dotenv()


EQUOS_AGENT_ID = os.getenv("EQUOS_AGENT_ID", "")
EQUOS_AVATAR_ID = os.getenv("EQUOS_AVATAR_ID", "")


def run_session() -> None:
    client = Equos(api_key=os.getenv("EQUOS_API_KEY", ""))

    response: CreateEquosSessionResponse = client.sessions.create(
        data=CreateEquosSessionRequest(
            name="Translation Session",
            client=None,  # Optional: specify a client ID if needed for resource segmentation.
            agent=EquosResourceId(id=EQUOS_AGENT_ID),
            avatar=EquosResourceId(id=EQUOS_AVATAR_ID),
            consumerIdentity=EquosParticipantIdentity(name="User", identity="user-id"),
        )
    )

    print("Session created:", response.session.id)
    print("Access token:", response.consumerAccessToken)

    session: Optional[EquosSession] = client.sessions.stop(id=response.session.id)

    if session:
        print("Session status:", session.status)


if __name__ == "__main__":
    run_session()
